examples/rebuild_shared_projection/src/main.rs

```diff
@@ -0,0 +1,91 @@
+use aggregate_merging::aggregates::{AggregateA, AggregateB};
+use aggregate_merging::projectors::{Counter, CounterProjector};
+use aggregate_merging::structs::{CommandA, CommandB, CounterError, ProjectorEvent};
+use esrs::aggregate::{AggregateManager, AggregateState};
+use esrs::projector::SqliteProjector;
+use esrs::store::StoreEvent;
+
+use sqlx::{pool::PoolOptions, Pool, Sqlite};
+use uuid::Uuid;
+
+// A simple example demonstrating rebuilding a read-side projection from an event
+// stream
+#[tokio::main]
+async fn main() {
+    let pool: Pool<Sqlite> = PoolOptions::new()
+        .connect("sqlite::memory:")
+        .await
+        .expect("Failed to create pool");
+
+    let count_id = Uuid::new_v4();
+
+    setup(&pool, count_id.clone()).await;
+    let agg_a = AggregateA::new(&pool).await.unwrap();
+    let agg_b = AggregateB::new(&pool).await.unwrap();
+    let store_a = agg_a.event_store();
+    let store_b = agg_b.event_store();
+
+    let ids = vec![count_id];
+    let projectors: Vec<Box<dyn SqliteProjector<ProjectorEvent, CounterError>>> = vec![Box::new(CounterProjector)];
+    for id in ids {
+        // For a given ID, retrieve the event lists for both aggregates
+        let events_a = store_a.by_aggregate_id(id).await.unwrap();
+        let events_b = store_b.by_aggregate_id(id).await.unwrap();
```

Comment:
Aggregate b has different aggregate id for sure. This approach of rebuilding stuff by id can work but with more complex logic, maybe looking for the aggregate id of aggregate b inside the payload where the key is linked. But it's definitely not possible to have two aggregates sharing same id